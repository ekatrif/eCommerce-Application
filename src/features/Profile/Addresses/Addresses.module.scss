.profile__addresses {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
  column-gap: 10px;
}

.profile__item {
  padding: 1rem;
  border-radius: 10px;
  background-color: #e9e7e7;

  &__title {
    font-size: 1.3rem;
  }

  &__value {
    margin-top: 10px;
  }

  &:first-child {
    border-top-left-radius: 0px;
  }

  .default {
    font-size: 0.8rem;
    width: fit-content;
    color: #29892d;
    border: 1px solid #29892d;
    border-radius: 5px;
    padding: 3px;
    margin-top: 5px;
  }
}

