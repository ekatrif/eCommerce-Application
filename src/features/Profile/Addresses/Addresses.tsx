import classes from './Addresses.module.scss';

interface AddressesProps {
  type: string;
  country: string;
  city: string;
  street: string;
  postCode: string;
}

const Addresses: React.FC<AddressesProps> = ({ type, country, city, street, postCode }) => {
  return (
    <div>
      <form className={classes.profile__addresses}>
        <label className={classes.profile__item}>
          <p className={classes.profile__item__title}>TYPE</p>
          <p className={classes.profile__item__value}>
            {type}
            <span className={classes.default}>default</span>
          </p>
        </label>
        <label className={classes.profile__item}>
          <p className={classes.profile__item__title}>COUNTRY</p>
          <p className={classes.profile__item__value}>{country}</p>
        </label>
        <label className={classes.profile__item}>
          <p className={classes.profile__item__title}>CITY</p>
          <p className={classes.profile__item__value}>{city}</p>
        </label>
        <label className={classes.profile__item}>
          <p className={classes.profile__item__title}>STREET</p>
          <p className={classes.profile__item__value}>{street}</p>
        </label>
        <label className={classes.profile__item}>
          <p className={classes.profile__item__title}>POSTAL CODE</p>
          <p className={classes.profile__item__value}>{postCode}</p>
        </label>
      </form>
    </div>
  );
};

export default Addresses;
