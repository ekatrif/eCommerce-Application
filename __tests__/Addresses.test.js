import React from 'react';
import renderer from 'react-test-renderer';
import Addresses from '../src/features/Profile/Addresses/Addresses';

test('Addresses renders correctly', () => {
  const tree = renderer
    .create(<Addresses />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
