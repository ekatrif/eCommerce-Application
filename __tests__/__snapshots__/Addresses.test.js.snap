// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addresses renders correctly 1`] = `
<div>
  <div
    className="profile__addresses"
  >
    <div
      className="profile__item type"
    >
      <p
        className="profile__item__title"
      >
        TYPE
      </p>
      <div
        className="profile__item__value"
      >
        <p>
          -
        </p>
        <div>
          <p
            className="default"
          >
            default billing
          </p>
          <p
            className="default"
          >
            default shipping
          </p>
        </div>
      </div>
    </div>
    <div
      className="profile__item"
    >
      <p
        className="profile__item__title"
      >
        COUNTRY
      </p>
      <p
        className="profile__item__value"
      />
    </div>
    <div
      className="profile__item"
    >
      <p
        className="profile__item__title"
      >
        CITY
      </p>
      <p
        className="profile__item__value"
      />
    </div>
    <div
      className="profile__item"
    >
      <p
        className="profile__item__title"
      >
        STREET
      </p>
      <p
        className="profile__item__value"
      />
    </div>
    <div
      className="profile__item postalCode"
    >
      <p
        className="profile__item__title"
      >
        POSTAL CODE
      </p>
      <p
        className="profile__item__value"
      />
    </div>
  </div>
</div>
`;
